packages/http/src/mocker/HttpMocker.ts

```diff
@@ -12,17 +12,26 @@ import {
   ProblemJsonError,
 } from '../types';
 import { UNPROCESSABLE_ENTITY } from './errors';
+import { generate, generateStatic } from './generator/JSONSchema';
 import helpers from './negotiator/NegotiatorHelpers';
 import { IHttpNegotiationResult } from './negotiator/types';
 
 export class HttpMocker implements IMocker<IHttpOperation, IHttpRequest, IHttpConfig, IHttpResponse> {
-  constructor(private _exampleGenerator: PayloadGenerator) {}
-
-  public async mock({
+  public mock({
     resource,
     input,
     config,
-  }: Partial<IMockerOpts<IHttpOperation, IHttpRequest, IHttpConfig>>): Promise<IHttpResponse> {
+  }: Partial<IMockerOpts<IHttpOperation, IHttpRequest, IHttpConfig>>): IHttpResponse {
+    let payloadGenerator: PayloadGenerator = generateStatic;
+
+    if (config) {
+      if (typeof config.mock !== 'boolean') {
```

Comment:
maybe you could use some && instead of 3 levels of nesting?

Reply:
5aed8b8